libs/ui/src/hooks/useHardware.test.ts

```diff
@@ -0,0 +1,310 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { Logger, LogSource, LogEventId } from '@votingworks/logging';
+import {
+  AccessibleControllerProductId,
+  AccessibleControllerVendorId,
+  FujitsuFi7160ScannerProductId,
+  FujitsuScannerVendorId,
+  MemoryHardware,
+  OmniKeyCardReaderProductId,
+  OmniKeyCardReaderVendorId,
+  PlustekScannerVendorId,
+  PlustekVTM300ScannerProductId,
+} from '@votingworks/utils';
+import { useHardware } from './useHardware';
+
+test('can connect printer as expected', async () => {
+  const hardware = new MemoryHardware();
+  const fakeLogger = new Logger(LogSource.VxBatchScanApp);
+  const logSpy = jest.spyOn(fakeLogger, 'log').mockResolvedValue();
+  const { result, rerender } = renderHook(() =>
+    useHardware({ hardware, logger: fakeLogger })
+  );
+  expect(result.current.hasPrinterAttached).toBe(false);
+  expect(logSpy).toHaveBeenCalledTimes(0);
+
+  await act(async () => await hardware.setPrinterConnected(true));
+  rerender();
+  expect(result.current.hasPrinterAttached).toBe(true);
+  expect(logSpy).toHaveBeenCalledTimes(2);
+  expect(logSpy).toHaveBeenNthCalledWith(
+    1,
+    LogEventId.PrinterAttached,
+    'system',
+    expect.objectContaining({
+      message: expect.stringContaining('New printer attached'),
+      connected: 'true',
+    })
+  );
+  expect(result.current.hasCardReaderAttached).toBe(false);
+  expect(result.current.hasFujitsuScannerAttached).toBe(false);
+  expect(result.current.hasPlustekScannerAttached).toBe(false);
+  expect(result.current.hasAccessibleControllerAttached).toBe(false);
```

Comment:
Grouping these into one assertion can often improve debugging since you'll get more information in the error message. Your call.

```suggestion
  expect(result.current).toEqual({
    hasCardReaderAttached: false,
    hasFujitsuScannerAttached: false,
    hasPlustekScannerAttached: false,
    hasAccessibleControllerAttached: false,
  });
```